packages/pwa-kit-dev/src/utils/extensibility-utils.test.js

```diff
@@ -0,0 +1,131 @@
+ */
+
+import fs from 'fs-extra'
+import path from 'path'
+
+import * as extensionUtils from './extensibility-utils'
+
+describe('extensibilityUtils', () => {
+    const existingTSFile = path.join('mocked', 'path', 'exists.ts')
+    const existingJSFile = path.join('mocked', 'path', 'exists.js')
+    const existingFileWithoutExtension = path.join('mocked', 'path', 'exists')
+    const missingFileWithoutExtension = path.join('mocked', 'path', 'does-not-exists')
+
+    beforeAll(() => {
+        const spy = jest.spyOn(fs, 'existsSync')
+
+        spy.mockImplementation((filePath) => {
+            if (filePath === existingJSFile) {
+                return true
+            }
+
+            if (filePath === existingTSFile) {
+                return true
+            }
+
+            if (
+                filePath ===
+                path.join(process.cwd(), 'node_modules', '@extension-good', 'src', 'setup-app.ts')
+            ) {
+                return true
+            }
+
+            return null
+        })
+    })
+
+    describe('"findFileWithExtension" correctly', () => {
+        ;[
+            {
+                name: 'returns null if path exists but no files no files with provided file extensions.',
+                path: existingFileWithoutExtension,
+                fileExtensions: ['.mjs'],
+                expected: null
+            },
+            {
+                name: 'returns path if file is found with provided extension.',
+                path: existingFileWithoutExtension,
+                fileExtensions: ['.js'],
+                expected: existingJSFile
+            },
+            {
+                name: 'returns path if file is found with mixed existing and non existing extensions.',
+                path: existingFileWithoutExtension,
+                fileExtensions: ['.mjs', '.ts'],
+                expected: existingTSFile
+            },
+            {
+                name: 'returns null if no file extensions are provided.',
+                path: existingFileWithoutExtension,
+                fileExtensions: undefined,
+                expected: null
+            },
+            {
+                name: 'returns null if no file extensions are provided.',
+                path: missingFileWithoutExtension,
+                fileExtensions: ['.js'],
+                expected: null
+            }
+        ].forEach((testCase) => {
+            test(`${testCase.name}`, () => {
+                const {path, fileExtensions} = testCase
+                const result = extensionUtils.findFileWithExtension(path, fileExtensions)
+
+                expect(result).toEqual(testCase.expected)
+            })
+        })
+    })
+
+    describe('"buildAliases" correctly', () => {
+        ;[
+            {
+                name: 'returns an empty object if undefined is passed',
+                extensions: [],
+                expected: {}
```

Comment:
In this case, we don't want to pass in anything for the `extensions`.
```suggestion
                name: 'returns an empty object if undefined is passed',
                expected: {}
```

Reply:
I've addressed this. Thanks